File: core/src/org/rexellentgames/dungeon/entity/item/accessory/equipable/LuckyCube.java
package org.rexellentgames.dungeon.entity.item.accessory.equipable;

import org.rexellentgames.dungeon.assets.Locale;
import org.rexellentgames.dungeon.entity.creature.player.Player;

public class LuckyCube extends Equipable {
	{
		name = Locale.get("lucky_cube");
		description = Locale.get("lucky_cube_desc");
		sprite = "item-damage_roll";
	}

	@Override
	public void onEquip() {
		super.onEquip();

		if (this.owner instanceof Player) {
			((Player) this.owner).luckDamage = true;
		}
	}

	@Override
	public void onUnequip() {
		super.onUnequip();

		if (this.owner instanceof Player) {
			((Player) this.owner).luckDamage = false;
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/LuckyCube.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

import org.rexcellentgames.burningknight.assets.Locale

class LuckyCube : Equippable() {
	val chance: Float
		get() = this.level * 5f + 45f

	init {
		name = Locale.get("lucky_cube")
		description = Locale.get("lucky_cube_desc")
		sprite = "item-damage_roll"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.luckDamage = true
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.luckDamage = false
	}

	override fun getMaxLevel(): Int {
		return 15
	}

	override fun getDescription(): String {
		return super.getDescription().replace("{CHANCE}", chance.toInt().toString())
	}
}